### PR TITLE
Aiohttp path params

### PR DESCRIPTION
**Pull request by Teyras**
- Support reading path parameters as view function arguments in AioHTTP
- Use :param directives in doc blocks to document parameters

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -62,6 +62,20 @@ web.run_app(app)
...
+
+```python
+from aiohttp import web
+from apistrap.aiohttp import AioHTTPApistrap
+
+oapi = AioHTTPApistrap()
+routes = web.RouteTableDef()
+
+@routes.get("/{param_a}/{param_b}")
+def endpoint(param_a: str, param_b: int):  # Note that the request parameter is optional
```
Comment (petrbel) [inline README.md]:
```suggestion
def endpoint(param_a: str, param_b: int = 5):  # Note that the request parameter is optional
```

Comment (petrbel) [reply README.md]: add example of default value

File: apistrap/aiohttp.py [hunk 2]
```diff
@@ -278,7 +284,68 @@ def _extract_specs(self) -> None:
...
+        If necessary, wrap the route handler with a coroutine that accepts a AioHTTP request, extracts parameters to
+        satisfy the underlying handler and forwards them to the original handler.
+
+        :param route: the route whose handler should be wrapped
+        """
+
+        signature = inspect.signature(route.handler)
+
+        request_param: Optional[inspect.Parameter] = next(
+            filter(lambda p: issubclass(p.annotation, BaseRequest), signature.parameters.values()), None
```
Comment (petrbel) [inline apistrap/aiohttp.py]: are we sure there's at most one request? do we want to assert that?

File: apistrap/aiohttp.py [hunk 3]
```diff
@@ -278,7 +284,68 @@ def _extract_specs(self) -> None:
...
+            not request_param
+            and "request" in signature.parameters.keys()
+            and signature.parameters["request"].annotation == inspect.Signature.empty
+        ):
+            request_param = signature.parameters["request"]
+
+        takes_aiohttp_request = request_param is not None
+
+        additional_params: [inspect.Parameter] = [
+            *filter(
```
Comment (petrbel) [inline apistrap/aiohttp.py]: btw isn't `list(filter(...))` more explicit than `[*filter(...)]`?

File: apistrap/aiohttp.py (hunk 2 above)
Comment (Teyras) [reply apistrap/aiohttp.py]: That's probably a good idea.

File: apistrap/aiohttp.py [hunk 4]
```diff
@@ -278,7 +284,68 @@ def _extract_specs(self) -> None:
...
+        if (
+            not request_param
+            and "request" in signature.parameters.keys()
+            and signature.parameters["request"].annotation == inspect.Signature.empty
+        ):
+            request_param = signature.parameters["request"]
+
+        takes_aiohttp_request = request_param is not None
+
+        additional_params: [inspect.Parameter] = [
```
[the remark this answers is not in the document]
Comment (Teyras) [reply apistrap/aiohttp.py]: Not really, but I thought it does. Will fix.

File: apistrap/aiohttp.py (hunk 3 above)
Comment (Teyras) [reply apistrap/aiohttp.py]: I think once you know the second form, it looks cleaner.

File: tests/test_aiohttp_docblocks.py [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+async def test_summary_from_docblock(app_with_params_as_args, aiohttp_initialized_client):
+    client = await aiohttp_initialized_client(app_with_params_as_args)
+    response = await client.get("/spec.json")
+
+    assert response.status == 200
+    data = await response.json()
+    path = data["paths"]["/{param_a}/{param_b}"]["get"]
+
+    assert path["summary"] == "A cool view handler."
```
[the remark this answers is not in the document]
Comment (Teyras) [reply tests/test_aiohttp_docblocks.py]: Yeah, we need those tests too... Along with finding a suitable library for parsing all of the stuff you can have in a docblock.

**Comment by Teyras** [comment]
@blazekadam what do you think?

**Pull request merged**